src/python/pants/pantsd/pants_daemon.py

```diff
@@ -336,11 +336,15 @@ def _pantsd_logging(self):
             init_rust_logger(self._log_level, self._log_show_rust_3rdparty)
             result = setup_logging(
                 self._log_level,
+                native=self._native,
                 log_dir=self._log_dir,
                 log_name=self.LOG_NAME,
-                native=self._native,
                 warnings_filter_regexes=self._bootstrap_options.for_global_scope(),
             )
+            assert result is not None, (
```

Comment:
Actually, the whole point of the @overload was to prove this can't crash. I had to type hint the `PantsDaemon` constructor to finish this thought, but the assert is now safely removed and instead performed by mypy.

Reply:
Fixed.